src/System.Text.Primitives/System/Text/Encoding/Utf8/Utf8Encoder.cs

```diff
@@ -56,6 +56,32 @@ public static bool TryDecode(ReadOnlySpan<byte> utf8, Span<char> utf16, out int
             bytesConsumed = 0;
             charactersWritten = 0;
 
+            if (Vector.IsHardwareAccelerated)
+            {
+                var utf8Vectors = utf8.NonPortableCast<byte, Vector<byte>>();
+                var utf16Vectors = utf16.NonPortableCast<char, Vector<ushort>>();
+
+                // note: might not have enough space in utf16 for everything
+                if (utf16Vectors.Length < utf8Vectors.Length * 2)
+                {
+                    utf8Vectors = utf8Vectors.Slice(0, utf16Vectors.Length / 2);
+                }
+
+                int utf16index = 0;
+                for (int i = 0; i < utf8Vectors.Length; i++)
+                {
+                    var bytes = utf8Vectors[i];
```

Comment:
The Span indexer is non-trivially slower than using the by-ref / Unsafe.Add method for extracting values. Do you have new numbers given your recent changes?